makefiles/build_extensions

```diff
@@ -0,0 +1,14 @@
+PROJ_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
```

Comment:
I think this makefile is really a bit of a hack, it should not live in a root dir of duckdb/duckdb called makefiles i think. I'm fine with sortof hacking this CI together to get started with moving to using the extension-ci-tools based workflow, but then i'd rather see the makefile live in the `.github` directory to make it clear that its intended for CI use only. Additionally I would like to see a comment in the makefile explaining what its for because its a bit hacky

Reply:
Thanks, had a brief explanation